File: 0038-count-and-say/0038-count-and-say.js
/**
 * @param {number} n
 * @return {string}
 */
var countAndSay = function(n) {
    if (n === 1) {
        return '1'
    }
    
    const countAndSayHelper = (val) => {
        strVal = '' + val
        let builder = ''
        let currNum = strVal[0]
        let currCount = 0
        for (let i = 0; i < strVal.length; i++) {
            if (strVal[i] === currNum) {
                currCount++
            } else {
                builder += currCount + '' + currNum
                currNum = strVal[i]
                currCount = 1
            }
        }
        builder += currCount + '' + currNum
        return builder
    }
    
    return countAndSayHelper(countAndSay(n - 1))
};

File: 0038-count-and-say/0038-count-and-say.py
class Solution(object):
    def countAndSay(self, n):
        """
        :type n: int
        :rtype: str
        """
        if n == 1:
            return '1'
        
        return countAndSayHelper(self.countAndSay(n - 1))
    
def countAndSayHelper(val):
    strVal = str(val)
    builder = ''
    currNum = strVal[0]
    currCount = 0
    for i in val:
        if i == currNum:
            currCount += 1
        else:
            builder += str(currCount) + str(currNum)
            currNum = i
            currCount = 1
    builder += str(currCount) + str(currNum)
    return builder